File: airline-booking-service/airline-booking-service/src/main/java/com/miu/flightmanagement/airlinebookingservice/model/ScheduledFlight.java
package com.miu.flightmanagement.airlinebookingservice.model;

import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;

@Entity
@Getter
@Setter
@ToString
@EqualsAndHashCode
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ScheduledFlight {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long scheduleFlightId;

	@ManyToOne
	private Flight flight;

	private Integer availableSeats;

	private Integer temporaryAvailableSeats;

	private BigDecimal economicPrice;

	@OneToOne(cascade = {CascadeType.PERSIST, CascadeType.MERGE})
	private Schedule schedule;
}

File: client/src/app/models/scheduledFlight.ts
import { Flight } from "./flight";
import { Schedule } from "./schedule";

export interface ScheduledFlight {
  scheduledFlightId: number;
  flight: Flight;
  availableSeats: number;
  schedule: Schedule;
  economicPrice: string | number;
}

export interface ScheduledFlights {
  scheduledFlights: ScheduledFlight[];
}
